src/game/GameState.ts

```diff
@@ -0,0 +1,76 @@
+import { Field, Struct, MerkleMap, PublicKey, UInt32 } from 'snarkyjs';
+
+import { Piece } from '../objects/Piece';
+import { Position } from '../objects/Position';
+import { Unit } from '../objects/Unit';
+
+export class GameState extends Struct({
+  piecesRoot: Field, // root hash of pieces in the arena keyed by their id
+  arenaRoot: Field, // root hash of a merkle map of positions which are occupied
+  playerTurn: Field,
+  player1: PublicKey,
+  player2: PublicKey,
+  arenaLength: UInt32,
+  arenaWidth: UInt32,
+  turnsCompleted: UInt32,
```

Comment:
Do we plan for `turnsCompleted` to be 1 or 2 when the first player starts their second turn? In other words, does this number increment each time a player completes a turn? Or at the end of each battle round (which consists of one turn by each player)?

Reply:
Probably the former.  The main thing is that we can't accept a valid turn that is too old.  Since this hasn't actually been implemented, the details aren't exactly nailed down.